plugins/in_tail/tail.c

```diff
@@ -106,6 +104,11 @@ static int in_tail_collect_pending(struct flb_input_instance *i_ins,
         }
     }
 
+    /* If no more active files, consume pending signal so we don't get called again. */
+    if (active == 0) {
+        tail_consume_pending(ctx);
+    }
```

Comment:
This change isn't necessary now that I made the pipe NONBLOCK. However, it should reduce the number of duplicate signals added to the pending pipe - so I think it's worth keeping.